contrib/utilities/indent.py

```diff
@@ -0,0 +1,260 @@
+
+from __future__ import print_function
+
+import argparse
+import time
+import distutils.spawn
+import distutils.file_util
+
+import fnmatch
+import multiprocessing
+import os
+import logging
+import Queue
+import re
+import shutil
+import subprocess
+import threading
+
+from tempfile import mkdtemp, mkstemp
+from filecmp  import cmp
+
+#
+# This script can be used to format code in external projects
+# (that using deal.II) by adding .clang_format files.
+#
+# This program formats code using 'clang-format'.
+# If this program is executed on dry run mode, only the file names of
+# the files that are not formatted correctly are reported
+# without actually formatting them.
+#
+# If this program is executed on dry run mode and if the code base is
+# correctly formatted, then the program doesn't write out anything.
+# This functionality can be exploited by a pre-commit git-hook
+# to check whether the codebase is formatted correctly.
+#
+# Usage:     python contrib/utilities/indent.py
+#
+
+def parse_arguments ():
+  """
+  Argument parser.
+  """
+  parser = argparse.ArgumentParser("Run clang-format on a list of files "
```

Comment:
maybe something like:

Run clang-format on all file in a list of directories that satisfy the regex.